WIP: #task2/add-bowling-game

satonin12 opened the pull request.

satonin12 committed:
ci(tests): ci only on branch #task2

ci auto start tests only in #task2 pushing on open MR

satonin12 committed:
test: add base test to return score strike + bonus (+ next 2 frame score)

satonin12 committed:
feat(bowling-class): add get methods score, isStrike, strikeBonus

satonin12 committed:
test(tests): add test to calc all zero & all frames 1 pin

satonin12 committed:
feat(bowling-class): add method isSpare, spareBonus & sumFrame

satonin12 committed:
test(tests): add tests to other combination bowling game

satonin12 committed:
refactor(bowling-class): refactor get score function to done tests

satonin12 committed:
refactor(bowling-class): refactor get score function to done tests

additional check that the strike was made after the spare

satonin12 committed:
refactor(bowling-class): add types for methods class

satonin12 committed:
test(tests): add tests for a get last score and throws

satonin12 committed:
feat(bowling-class): add geter method for a get last frame & throws

get in a format {score: n, rolls: [...n]}

satonin12 committed:
refactor(tests): delete console.log base test, add description

satonin12 committed:
refactor(bowling-class): delete commentary, logToConsole method

Koval1997 commented on src/Bowling/Bowling.ts:
```diff
@@ -1,7 +1,64 @@
+  // methods
+  roll = (pins : number) => this.frames.push(pins);
+
+  get score() {
```
score() и lastScore() сделаны как геттеры. Геттеры и сеттеры все-таки использовать лучше для свойств, а эти штуки лучше описать как функции.

Koval1997 commented on src/Bowling/Bowling.ts:
```diff
@@ -1,7 +1,64 @@
+    return score;
+  };
+
+  get lastScore() {
```
score() и lastScore() сделаны как геттеры. Геттеры и сеттеры все-таки использовать лучше для свойств, а эти штуки лучше описать как функции.

Koval1997 commented on src/services/services.ts:
```diff
@@ -0,0 +1,6 @@
+export const checkNumber = (number : number | null | undefined) : number => {
```
Такие вот мелкие штуки, которые проверяют форма числа, парсят строку и т.д. принято класть в папку Utils. В данном случае checkNumber, возвращает 0, если входящий параметр не falsy значение. Не очень понял, зачем этот метод вообще, вряд ли очки за бросок могут быть null или undefined и т.д. Надо стараться избегать проверок на null и т.д. в тех местах, где их по логике быть не должно :)

Koval1997 commented on src/Bowling/Bowling.ts:
```diff
@@ -1,7 +1,64 @@
+      score,
+      rolls,
+    };
+  }
```
можно заменить на lastScore() {
    return {
      score: this.score,
      rolls: this.frames,
    };
  }

satonin12 committed:
fix(bowling-class-tests): score, get score to func

score, lastScore getter => function & refactor place where are they called, add them to types
